### PR TITLE
Update generate_template function and add UI for incomplete file

### LINKED ISSUE #635: Update generate_template function and add UI for incomplete file
# Description

~~Add a `draft` key in the meta of a thesaurus to add a draft version of the thesaurus.~~

A new command `generate_missing_template` executes the command `generate_template` for each missing thesaurus.

~~Also, add the argument `draft` for the command `generate_template` and function `generate_language_template`.~~

And finally, an alert on "compare" and "reference" warns about the draft version of the thesaurus with a link to GitHub.

~~PS: To summarize the use of "draft" mode, this can be used to identify the incomplete thesaurus.~~

# Requirements

A PR will be available with these features.

# Additional Notes

image

### PR DESCRIPTION
**Pull request by ulyssear**
## What GitHub issue does this PR apply to?

Resolves #635 

## What changed and why?

   * New command `generate_missing_templates` which generates all missing thesaurus.
     ```bash
     python manage.py generate_missing_templates
     ```

   * New function in the `Language` model  named `is_incomplete`.
     ```python
     def is_incomplete(self, version, structure_key):
        """
        Returns a Boolean if the version is incomplete

        :param version: the version of the language
        :param structure_key: the ID for the structure to load
        :return: Boolean if the version is incomplete
        """

        template = self.load_filled_concepts(structure_key, version)
        template = json.loads(template)
        
        for concept in template["concepts"].values():
            code = concept.get("code")
            not_implemented = concept.get("not-implemented", None)
            comment = concept.get("comment", None)
            if isinstance(code, list):
                code = "".join(code)
            if isinstance(code, str) and code.strip() == "" and not_implemented is None and comment is None:
                return True

        return False
     ```

## (If editing Django app) Please add screenshots

image

## Checklist

- [ ] I claimed any associated issue(s) and they are not someone else's
- [x] I have looked at the documentation to ensure I made any revisions correctly
- [x] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

## Any additional comments or things to be aware of while reviewing?

### CODE REVIEW

**Comment by ulyssear** [comment]
I'm working on unit tests, but I have no idea how I could add draft mode inside the tests.
Would it be possible to have a track of what I could put?

File: web/thesaurus_template_generators.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 from web.models import MetaInfo
 
 
-def generate_language_template(language_id, structure_id, version=None):
+def generate_language_template(language_id, structure_id, version=None, draft=False):
```
Comment (geekygirlsarah) [inline web/thesaurus_template_generators.py]: Should this get removed as I think we decided against "draft mode"?

File: web/thesaurus_template_generators.py [hunk 2]
```diff
@@ -20,6 +20,9 @@ def generate_language_template(language_id, structure_id, version=None):
 
     if version:
         meta['language_version'] = version
+    
+    if draft:
+        meta['draft'] = True
```
Comment (geekygirlsarah) [inline web/thesaurus_template_generators.py]: Same for removing this

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]
So besides the two comments, when I go to the [review app](https://codethesaurus-test-pr-636.herokuapp.com/reference/?concept=functions&lang=ada%3B2012), I don't see the UI addition. Shouldn't I see it here?

image

File: web/thesaurus_template_generators.py (hunk 1 above)
Comment (ulyssear) [reply web/thesaurus_template_generators.py]: I forgot to remove it, sorry for that.

**Comment by ulyssear** [comment]
> So besides the two comments, when I go to the [review app](https://codethesaurus-test-pr-636.herokuapp.com/reference/?concept=functions&lang=ada%3B2012), I don't see the UI addition. Shouldn't I see it here?
> 
> image

OK, I know what's happened: the method "is_incomplete" checks if the code fields are empty in the template inside the thesaurus folder. But in the case of some files, these entries (with "Unknown") are not present and thus, no condition is done there (so, the result will be False, instead of True).
We have to check with the complete template but I don't know if it's a good practice, I will even so put this solution.

**Comment by geekygirlsarah** [comment]
I'm going to go ahead and close this because there's some merge conflicts and I haven't seen activity since November.

If you want to keep working on it, you're welcome to! You can reopen this or make a new pull request to keep working on it.

Thanks for the work you've put into it so far!

**Comment by ulyssear** [comment]
I'll check on my side to fix the merge conflicts !
I may do this this week, or next week at the latest.

**Pull request closed without merging**